### PR TITLE
Add missing types on chunk related classes

### PR DESCRIPTION
**Pull request by ooflorent**
**What kind of change does this PR introduce?**

bugfix and types

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

nothing

### CODE REVIEW

File: lib/ChunkGroup.js [hunk 1]
```diff
@@ -290,16 +320,20 @@ class ChunkGroup {
...
 	}
 
+	/**
+	 * @param {ChunkGroup} chunkGroup the parent group
+	 * @returns {boolean} returns true if this group has been removed from the parent
+	 */
 	removeParent(chunkGroup) {
 		if (this._parents.delete(chunkGroup)) {
-			chunkGroup.removeChunk(this);
+			chunkGroup.removeChild(this);
```
Comment (ooflorent) [inline lib/ChunkGroup.js]: I think this was a bug:

- `ChunkGroup#removeChunk` if for `Chunk` ❌
- `ChunkGroup#removeChild` is for `ChunkGroup`

File: lib/Compilation.js [hunk 2]
```diff
@@ -1917,7 +1917,6 @@ class Compilation {
 			for (let indexChunk = 0; indexChunk < chunks.length; indexChunk++) {
 				const iteratedChunk = chunks[indexChunk];
 				chunkGroup.removeChunk(iteratedChunk);
-				chunkGroup.removeParent(iteratedChunk);
```
Comment (ooflorent) [inline lib/Compilation.js]: Another bug here. `ChunkGroup#removeParent` is for `ChunkGroup`, not chunks.

File: lib/ChunkGroup.js (hunk 1 above)
Comment (sokra) [reply lib/ChunkGroup.js]: :+1:

File: lib/Compilation.js (hunk 2 above)
Comment (sokra) [reply lib/Compilation.js]: :+1:

**Comment by sokra** [comment]
Thanks

**Pull request merged**